src/segment-transmuxer.js

```diff
@@ -187,30 +187,30 @@ export const processTransmux = ({
   transmuxer.postMessage({ action: isPartial ? 'partialFlush' : 'flush' });
 };
 
-export const dequeue = () => {
-  currentTransmux = null;
-  if (transmuxQueue.length) {
-    currentTransmux = transmuxQueue.shift();
-    if (typeof currentTransmux === 'function') {
-      currentTransmux();
+export const dequeue = (transmuxer) => {
+  transmuxer.currentTransmux = null;
+  if (transmuxer.transmuxQueue && transmuxer.transmuxQueue.length) {
```

Comment:
Does this need to check for `transmuxer.transmuxQueue`?

Reply:
nope, I think I missed this one when I made `createTransmuxer` something that we use in tests and segment-transmuxer. Originally I had null checks everywhere but if we always rely on `createTransmuxer` from this file then it will always exist.i